packages/medusa/src/services/cart.ts

```diff
@@ -1631,12 +1632,11 @@ class CartService extends TransactionBaseService {
   }
 
   /**
-   * Sets a payment method for a cart.
+   * Selects a payment session for a cart and creates a payment object in the external provider system
    * @param cartId - the id of the cart to add payment method to
    * @param providerId - the id of the provider to be set to the cart
-   * @return result of update operation
    */
-  async setPaymentSession(cartId: string, providerId: string): Promise<Cart> {
+  async setPaymentSession(cartId: string, providerId: string): Promise<void> {
```

Comment:
self-review: I wanted to break down this method in smaller functions. Unfortunately I encounter a naming issue (in finding proper naming) as we have other session functions that would collide with the wanted naming. Therefore I keep that task for another time as I think we should have a separate service for cart payment session management.